realm/realm-library/src/main/java/io/realm/RealmBaseRecyclerAdapter.java

```diff
@@ -0,0 +1,195 @@
+package io.realm;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+
+/**
+ * The RealmBaseRecyclerAdapter class is an abstract utility class for binding RecyclerView UI elements to Realm data.
+ * <p>
+ * This adapter will automatically handle any updates to its data and call {@link #notifyDataSetChanged()} as
+ * appropriate.
+ * <p>
+ * The RealmAdapter will stop receiving updates if the Realm instance providing the {@link OrderedRealmCollection} is
+ * closed.
+ *
+ * @param <T> type of {@link RealmObject} stored in the adapter.
+ * @param <VH> type of {@link RecyclerView.ViewHolder} used in the adapter.
+ */
+public abstract class RealmBaseRecyclerAdapter<T extends RealmObject, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {
+
+    protected LayoutInflater inflater;
+    protected Context context;
+    private OrderedRealmCollection<T> adapterData;
+    private RealmChangeListener listener;
+
+    private boolean hasAutoUpdates;
+
+    public RealmBaseRecyclerAdapter(Context context, OrderedRealmCollection<T> data, boolean autoUpdate) {
+        if (context == null) {
+            throw new IllegalArgumentException("Context cannot be null");
+        }
+
+        this.context = context;
+        this.adapterData = data;
+        this.inflater = LayoutInflater.from(context);
+        this.hasAutoUpdates = autoUpdate;
+        if (autoUpdate) {
+           listener = new RealmChangeListener() {
+               @Override
+               public void onChange() {
+                   notifyDataSetChanged();
+               }
+           };
+        }
+    }
+
+    @Override
+    public void onAttachedToRecyclerView(final RecyclerView recyclerView) {
+        super.onAttachedToRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            addListener(adapterData);
+        }
+    }
+
+    @Override
+    public void onDetachedFromRecyclerView(final RecyclerView recyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView);
+        if (hasAutoUpdates && isDataValid()) {
+            removeListener(adapterData);
+        }
+    }
+
+    /**
+     * Returns the current ID for an item. Note that item IDs are not stable so you cannot rely on the item ID being the
+     * same after {@link #notifyDataSetChanged()} or {@link #updateData(OrderedRealmCollection)} has been called.
+     *
+     * @param index position of item in the adapter.
+     * @return current item ID.
+     */
+    @Override
+    public long getItemId(final int index) {
+        return index;
+    }
+
+    @Override
+    public int getItemCount() {
+        return isDataValid() ? adapterData.size() : 0;
+    }
+
+    /**
+     * Returns the item associated with the specified position.
+     *
+     * @param index index of the item.
+     * @return the item at the specified position.
+     */
+    public T getItem(int index) {
```

Comment:
This is also not part of the public interface for RecyclerView adapter. I can see why it would be a nice utility method though. So I probably think it is okay to keep it. What do you think @realm/java ?